lib/manager/composer/artifacts.ts

```diff
@@ -48,62 +126,12 @@ export async function updateArtifacts({
     if (config.isLockFileMaintenance) {
       await deleteLocalFile(lockFileName);
     }
-    const authJson = {};
-    let credentials = hostRules.find({
-      hostType: PLATFORM_TYPE_GITHUB,
-      url: 'https://api.github.com/',
-    });
-    // istanbul ignore if
-    if (credentials && credentials.token) {
-      authJson['github-oauth'] = {
-        'github.com': credentials.token,
-      };
-    }
-    credentials = hostRules.find({
-      hostType: PLATFORM_TYPE_GITLAB,
-      url: 'https://gitlab.com/api/v4/',
-    });
-    // istanbul ignore if
-    if (credentials && credentials.token) {
-      authJson['gitlab-token'] = {
-        'gitlab.com': credentials.token,
-      };
-    }
-    try {
-      // istanbul ignore else
-      if (is.array(config.registryUrls)) {
-        for (const regUrl of config.registryUrls) {
-          if (regUrl) {
-            const { host } = URL.parse(regUrl);
-            const hostRule = hostRules.find({
-              hostType: datasourcePackagist.id,
-              url: regUrl,
-            });
-            // istanbul ignore else
-            if (hostRule.username && hostRule.password) {
-              logger.debug('Setting packagist auth for host ' + host);
-              authJson['http-basic'] = authJson['http-basic'] || {};
-              authJson['http-basic'][host] = {
-                username: hostRule.username,
-                password: hostRule.password,
-              };
-            } else {
-              logger.debug('No packagist auth found for ' + regUrl);
-            }
-          }
-        }
-      } else if (config.registryUrls) {
-        logger.warn(
-          { registryUrls: config.registryUrls },
-          'Non-array composer registryUrls'
-        );
-      }
-    } catch (err) /* istanbul ignore next */ {
-      logger.warn({ err }, 'Error setting registryUrls auth for composer');
-    }
+
+    const authJson = generateAuthJson(config);
     if (authJson) {
```

Comment:
This was always `true` as authJson was always an object. The new code will return null and therefore not write an empty auth.json files.